daml-script/dump/src/main/scala/com/daml/script/dump/Encode.scala

```diff
@@ -0,0 +1,253 @@
+        Doc.stack(r.fields.map(f => encodeField(partyMap, cidMap, f)))).nested(2)
+    }
+  }
+
+  private def encodeField(
+      partyMap: Map[String, String],
+      cidMap: Map[String, String],
+      field: RecordField,
+  ): Doc =
+    Doc.text(field.label) + Doc.text(" = ") + encodeValue(partyMap, cidMap, field.getValue.sum)
+
+  private def encodeParty(partyMap: Map[String, String], s: String): Doc = Doc.text(partyMap(s))
+
+  private def encodeParties(partyMap: Map[String, String], ps: Iterable[String]): Doc =
+    Doc.text("[") +
+      Doc.intercalate(Doc.text(", "), ps.map(encodeParty(partyMap, _))) +
+      Doc.text("]")
+
+  private def encodeCid(cidMap: Map[String, String], cid: String): Doc = {
+    // LedgerStrings are strings that match the regexp ``[A-Za-z0-9#:\-_/ ]+
+    Doc.text(cidMap(cid))
+  }
+
+  private def qualifyId(id: Identifier): Doc =
+    Doc.text(id.moduleName) + Doc.text(".") + Doc.text(id.entityName)
+
+  private def encodeEv(
+      partyMap: Map[String, String],
+      cidMap: Map[String, String],
+      ev: TreeEvent.Kind,
+  ): Doc = ev match {
+    case TreeEvent.Kind.Created(created) =>
+      Doc.text("createCmd ") + encodeRecord(partyMap, cidMap, created.getCreateArguments)
+    case TreeEvent.Kind.Exercised(exercised @ _) =>
+      Doc.text("exerciseCmd ") + encodeCid(cidMap, exercised.contractId) + Doc.space + encodeValue(
+        partyMap,
+        cidMap,
+        exercised.getChoiceArgument.sum,
+      )
+    case TreeEvent.Kind.Empty => throw new IllegalArgumentException("Unknown tree event")
+  }
+
+  private def bindCid(cidMap: Map[String, String], c: CreatedContract): Doc = {
+    Doc.text("let ") + encodeCid(cidMap, c.cid) + Doc.text(" = createdCid @") +
+      qualifyId(c.tplId) + Doc.text(" [") + Doc.intercalate(
+        Doc.text(", "),
+        c.path.map(encodeSelector(_)),
+      ) + Doc.text("] tree")
+  }
+
+  private def encodeTree(
+      partyMap: Map[String, String],
+      cidMap: Map[String, String],
+      tree: TransactionTree,
+  ): Doc = {
+    val rootEvs = tree.rootEventIds.map(tree.eventsById(_).kind)
+    val submitters = rootEvs.flatMap(evParties(_)).toSet
+    val cids = treeCids(tree)
+    (Doc.text("tree <- submitTreeMulti ") + encodeParties(partyMap, submitters) + Doc.text(
+      " [] do"
+    ) /
+      Doc.stack(rootEvs.map(ev => encodeEv(partyMap, cidMap, ev)))).hang(2) /
+      Doc.stack(cids.map(bindCid(cidMap, _)))
+  }
+
+  private def encodeSelector(selector: Selector): Doc = Doc.str(selector.i)
+
+  private def encodeImport(moduleName: String) =
+    Doc.text("import qualified ") + Doc.text(moduleName)
+
+  private def partyMapping(parties: Set[String]): Map[String, String] = {
+    // - PartyIdStrings are strings that match the regexp ``[A-Za-z0-9:\-_ ]+``.
+    def safeParty(p: String) =
+      Seq(":", "-", "_", " ").foldLeft(p) { case (p, x) => p.replace(x, "") }.toLowerCase
+    // Map from original party id to DAML identifier
```

Comment:
```suggestion
    // Map from original party id to Daml identifier
```

😛

Reply:
In my defense, this code is older than the rebranding!